scipy/linalg/special_matrices.py

```diff
@@ -1194,3 +1194,66 @@ def fiedler_companion(a):
     c[[0, 1], 0] = [-a[1], 1]
 
     return c
+
+def convmtx(a, n, mode='full'):
+    """
+    Construct a convolution matrix.
+
+    Constructs a matrix
+    `A = convmtx(a, n[, mode])`
+    such that
+    `np.dot(A, v)` is equivalent to `convolve(a, v[, mode])` but slower.
```

Comment:
I wonder if this is maybe a little challenging to read because i.e., `v` is defined in another docstring in NumPy. Maybe that's obvious to experts though.

Reply:
Good point.  I'll add the mathematical definition.

Reply:
I've put both `A @ v` and `matmul(A,v)` into the docstring.  The former is more modern, the latter is more descriptive and easier to research.